From 5e2ca1a88af9a817f1bda54961bfd6fb5ca19dd7 Mon Sep 17 00:00:00 2001
From: Christian Hohnstaedt <Christian.Hohnstaedt@wago.com>
Date: Mon, 7 Oct 2019 13:09:04 +0200
Subject: [PATCH] rlb: fix rlb_pde_switch_tx_state() for disabling

calling this function always enabled the tx_state,
independent of the provided value.

This is fixed now.

Signed-off-by: Christian Hohnstaedt <Christian.Hohnstaedt@wago.com>
---
 drivers/char/rmd.c | 4 ++--
 1 file changed, 2 insertions(+), 2 deletions(-)

diff --git a/drivers/char/rmd.c b/drivers/char/rmd.c
index bd1331c..7543d187 100644
--- a/drivers/char/rmd.c
+++ b/drivers/char/rmd.c
@@ -848,8 +848,8 @@ static long rmd_ioctl(struct file *filep, unsigned int cmd, unsigned long arg)
 		return rmd_wait_for_next_cycle(info->rmd, (void __user *)arg);
 
 	case RMD_ACTIVATE_TX_PATH:
-		info->is_tx_enabled = true;
-		return 0;
+		info->is_tx_enabled = arg ? 1 : 0;
+		return info->is_tx_enabled;
 
 	case RMD_GET_READ_TRIPPLE_BUF_CTRL_OFFSET:
 		return info->rx_channel.tripple_buf_ctrl_offset;
-- 
2.7.4

